torch/_subclasses/meta_utils.py

```diff
@@ -380,19 +426,102 @@ def is_c_of_r(complex_dtype, real_dtype):
+                                            None
+                                            if dynamic_dims is None
+                                            else dynamic_dims[1:]
+                                        ),
+                                        constraint_dims=(
+                                            None
+                                            if constraint_dims is None
+                                            else constraint_dims[1:]
+                                        ),
+                                    )
+                                    new_values = base.as_strided(
+                                        values_sizes,
+                                        values_strides,
+                                        values_storage_offset,
+                                    )
+
+                                    # Meta-ify other attributes (e.g. offsets, lengths)
+                                    attrs, ctx = t.__tensor_flatten__()
+                                    transformed_tensors_dict = {}
+                                    for attr in attrs:
+                                        inner_t = getattr(t, attr)
+                                        if attr == "_values":
+                                            continue
+                                        transformed_tensors_dict[attr] = callback(
+                                            lambda: empty_create(
+                                                inner_t,
+                                                AttrSource(source, attr),  # type: ignore[arg-type]
+                                                dynamic_dims=None,
+                                                constraint_dims=None,
+                                            )
+                                        )
+                                    new_offsets = transformed_tensors_dict["_offsets"]
+                                    new_lengths = transformed_tensors_dict.get(
+                                        "_lengths", None
+                                    )
+
+                                    # Wrap ragged size singleton symint to make it symbolic
+                                    ragged_size = get_symbolic_ragged_size(t, source)
+
+                                    # We're not calling __tensor_unflatten__() since we need a
+                                    # view, but we still want to ensure the fake offsets or
+                                    # lengths are associated with a symbolic ragged size.
+                                    ragged_source = (
+                                        new_offsets
+                                        if new_lengths is None
+                                        else new_lengths
+                                    )
+                                    _tensor_symint_registry[ragged_source] = ragged_size
```

Comment:
Do we need to manually create symbolic ragged size here anymore? Can't we rely on the `nested_view_from_values_offsets` function do do this for us since it's being called with fake tensors?

Reply:
Oh nvm I was being dumb lol

Reply:
Fixed by reducing code duplication